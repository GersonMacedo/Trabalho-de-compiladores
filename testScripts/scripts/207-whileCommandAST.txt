! -s TREE
! should print correctly the AST with the if else commands
program p ;
    var b1 : boolean;
    var b2 : boolean;
    var b3 : boolean;
    var i1 : integer;
    var i2 : integer;
    var i3 : integer;
    var i4 : integer;
begin
    while b1 do
        i1 := i2
    ;

    while b2 do
        while b3 do
            i3 := i4
    ;
end .