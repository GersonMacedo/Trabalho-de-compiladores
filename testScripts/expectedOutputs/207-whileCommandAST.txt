
Performing up to syntactic analysis and printing the AST tree at testScripts/scripts/207-whileCommandAST.txt
No errors found

Programa 'p'
Declarações:
    'b1': BOOLEAN
    'b2': BOOLEAN
    'b3': BOOLEAN
    'i1': INTEGER
    'i2': INTEGER
    'i3': INTEGER
    'i4': INTEGER
Corpo:
    while
        'b1'
    do
        'i1' :=
            'i2'
    while
        'b2'
    do
        while
            'b3'
        do
            'i3' :=
                'i4'
